### PR TITLE
[cli] support Thread Administrator Passcode configuration

### PR DESCRIPTION
**Pull request by librasungirl**

### CODE REVIEW

File: src/app/cli/interpreter.cpp [hunk 1]
```diff
@@ -912,17 +914,37 @@ Interpreter::Value Interpreter::ProcessConfig(const Expression &aExpr)
...
+        if (aExpr[2] == "accesscode")
+        {
+            mThreadAccessPasscode = aExpr[3];
+
+            // TODO(rongli@): add len and last digit check per SPEC-1216v11
+
+            for (char c : mThreadAccessPasscode)
+            {
+                pskc.push_back(static_cast<uint8_t>(c));
+            }
```
Comment (wgtdkp) [inline src/app/cli/interpreter.cpp:942]: nit:
```suggestion
            pskc.assign(mThreadAccessCode.begin(), mThreadAccessCode.end());
```

**Review by wgtdkp — APPROVED** [review]
Thanks, LGTM!

File: src/app/cli/interpreter.cpp [hunk 2]
```diff
@@ -197,7 +197,9 @@ const std::map<std::string, Interpreter::Evaluator> &Interpreter::mEvaluatorMap
 };
 
 const std::map<std::string, std::string> &Interpreter::mUsageMap = *new std::map<std::string, std::string>{
-    {"config", "config get pskc\n"
+    {"config", "config get accesscode\n"
+               "config set accesscode <9-digits>\n"
```
Comment (lanyuwen) [inline src/app/cli/interpreter.cpp:201]: not sure if I have some misunderstanding here, but get/set accesscode seems somewhat confusing to me, for which people might think it to get/set accesscode of border router. What about adding it as a parameter to get epskc? e.g., config get epskc -c <9-digits accesscode>?

Comment (librasungirl) [reply src/app/cli/interpreter.cpp]: updated with more friendly hints, pls take a look at the new patch

**Review by lanyuwen — APPROVED** [review]
LGTM

**Review by wgtdkp — APPROVED** [review]

File: src/app/cli/interpreter.cpp [hunk 3]
```diff
@@ -197,7 +197,9 @@ const std::map<std::string, Interpreter::Evaluator> &Interpreter::mEvaluatorMap
 };
 
 const std::map<std::string, std::string> &Interpreter::mUsageMap = *new std::map<std::string, std::string>{
-    {"config", "config get pskc\n"
+    {"config", "config get accesscode\n"
+               "config set accesscode <9-digits-thread-access-passcode>\n"
```
Comment (wgtdkp) [inline src/app/cli/interpreter.cpp:201]: Just come to realize that if the access code has the empty spaces removed, then the string is effectively a epskc. So probably we should change this command to "config set epskc"?

Comment (librasungirl) [reply src/app/cli/interpreter.cpp:201]: Offline chat with @wgtdkp, fine to keep it consistent with the terminology with Thread Specification.

File: src/app/cli/interpreter.cpp [hunk 4]
```diff
@@ -197,7 +197,9 @@ const std::map<std::string, Interpreter::Evaluator> &Interpreter::mEvaluatorMap
 };
 
 const std::map<std::string, std::string> &Interpreter::mUsageMap = *new std::map<std::string, std::string>{
-    {"config", "config get pskc\n"
+    {"config", "config get accesscode\n"
```
Comment (jwhui) [inline src/app/cli/interpreter.cpp:200]: nit: Thread Group TC call yesterday decided to call this "Thread Administrator Passcode".

Comment (librasungirl) [reply src/app/cli/interpreter.cpp]: rename in new patch

**Review by jwhui — APPROVED** [review]

**Pull request merged**